nautobot/core/forms/fields.py

```diff
@@ -714,6 +716,78 @@ def has_changed(self, initial, data):
         return super().has_changed(initial, data)
 
 
+class JSONArrayChoiceFormField(JSONArrayFormField):
+    """
+    A FormField counterpart to JSONArrayField that supports choice validation.
+    """
+
+    hidden_widget = MultipleHiddenInput
+    widget = widgets.StaticSelect2Multiple()
+    default_error_messages = {
+        "invalid_choice": _("Select a valid choice. %(value)s is not one of the available choices."),
+        "invalid_list": _("Enter a list of values."),
+    }
+
+    def __init__(self, base_field, choices, *, delimiter=",", **kwargs):
+        self.choices = choices
+        super().__init__(base_field, delimiter=delimiter, **kwargs)
+
+    # TODO: change this when we upgrade to Django 5, it uses a getter/setter for choices
+    def _get_choices(self):
+        return self._choices
+
+    def _set_choices(self, value):
+        if callable(value):
+            value = CallableChoiceIterator(value)
+        else:
+            value = list(value)
+        self._choices = self.widget.choices = value
+
+    choices = property(_get_choices, _set_choices)
+
+    def prepare_value(self, value):
+        if isinstance(value, list):
+            return value
+        return []
```

Comment:
Interesting that this is different from `JSONArrayField` in that the base class returns a string from `prepare_value` while this returns a list - why the difference?

Reply:
This is to support the StaticSelect2Multiple, it expects a list to exist on the data. If this is removed, then you can still select choices, but if you try to edit the form and there are two or more existing choices, the field show blank and would return null if you saved.